packaging/gce-disk-expand.spec

```diff
@@ -18,38 +18,38 @@ Version: %{_version}
 Release: g1
 License: Apache Software License
 Group: System Environment/Base
-URL: https://github.com/GoogleCloudPlatform/compute-image-packages
+URL: https://github.com/GoogleCloudPlatform/guest-diskexpand
 Source0: %{name}_%{version}.orig.tar.gz
 Requires: e2fsprogs, dracut, grep, util-linux, parted, gdisk
 Conflicts: dracut-modules-growroot
 
-BuildRequires: rsync
-
-# Allow other files in the source that don't end up in the package.
-%define _unpackaged_files_terminate_build 0
-
 %description
 This package resizes the root partition on first boot using parted.
 
 %prep
 %autosetup
 
 %install
-mv src/expandfs-lib.sh src/usr/share/dracut/modules.d/50expand_rootfs/
-%if 0%{?rhel} >= 7
-  ./dracut6_7.sh
-%endif
-rsync -Pravz src/ %{buildroot}
+mkdir -p %{buildroot}/usr/lib/dracut/modules.d/50expand_root
+cp -R src/usr/lib/dracut/modules.d/50expand_root %{buildroot}/usr/lib/dracut/modules.d/
+cp src/expandroot-lib.sh %{buildroot}/usr/lib/dracut/modules.d/50expand_root/
+mkdir -p %{buildroot}/usr/lib/systemd/system
+cp google-disk-expand.service %{buildroot}/usr/lib/systemd/system/
+mkdir -p %{buildroot}/usr/bin
+cp src/usr/bin/google_disk_expand %{buildroot}/usr/bin/
 
 %files
-%if 0%{?rhel} >= 7
- %attr(755,root,root) /usr/lib/dracut/modules.d/50expand_rootfs/*
-%else
- %attr(755,root,root) /usr/share/dracut/modules.d/50expand_rootfs/*
-%endif
+%attr(755,root,root) /usr/bin/google_disk_expand
+%attr(755,root,root) /usr/lib/dracut/modules.d/50expand_root/*
+%attr(644,root,root) /usr/lib/systemd/system/google-disk-expand.service
 
 %post
+# On initial install, not upgrade.
```

Comment:
Technically this happens on both install and upgrade (which is what we want).

Reply:
yeah i was going to limit to initial and decided against it, forgot to prune comment